steam_deploy.sh

```diff
@@ -0,0 +1,130 @@
+  eval "currentDepotPath=\$depot${i}Path"
+  if [ -n "$currentDepotPath" ]; then
+    currentDepot=$((appId+i))
+    echo ""
+    echo "Adding depot${currentDepot}.vdf ..."
+    echo ""
+    export DEPOTS="$DEPOTS  \"$currentDepot\" \"depot${currentDepot}.vdf\"\n  "
+    cat << EOF > "depot${currentDepot}.vdf"
+"DepotBuildConfig"
+{
+  "DepotID" "$currentDepot"
+  "FileMapping"
+  {
+    "LocalPath" "./$currentDepotPath/*"
+    "DepotPath" "."
+    "recursive" "1"
+  }
+  "FileExclusion" "*.pdb"
+}
+EOF
+
+  cat depot${currentDepot}.vdf
+  echo ""
+  fi;
+
+  i=$((i+1))
+done
+
+echo ""
+echo "#################################"
+echo "#    Generating App Manifest    #"
+echo "#################################"
+echo ""
+
+mkdir BuildOutput
+
+cat << EOF > "manifest.vdf"
+"appbuild"
+{
+  "appid" "$appId"
+  "desc" "$buildDescription"
+  "buildoutput" "BuildOutput"
+  "contentroot" "$(pwd)/$rootPath"
+  "setlive" "$releaseBranch"
+
+  "depots"
+  {$(echo "$DEPOTS" | sed 's/\\n/\
+/g')}
+}
+EOF
+
+cat manifest.vdf
+echo ""
+
+echo ""
+echo "#################################"
+echo "#    Copying SteamGuard Files   #"
+echo "#################################"
+echo ""
+
+mkdir -p "$STEAM_HOME/config"
+mkdir -p "/home/runner/Steam/config"
+
+echo "Copying $STEAM_HOME/config/config.vdf..."
+echo "$configVdf" > "$STEAM_HOME/config/config.vdf"
+chmod 777 "$STEAM_HOME/config/config.vdf"
+
+echo "Copying /home/runner/Steam/config/config.vdf..."
+echo "$configVdf" > "/home/runner/Steam/config/config.vdf"
+chmod 777 "/home/runner/Steam/config/config.vdf"
+
+echo "Copying $STEAM_HOME/ssfn..."
+echo "$ssfnFileContents" | base64 -d > "$STEAM_HOME/$ssfnFileName"
+chmod 777 "$STEAM_HOME/$ssfnFileName"
+
+echo "Copying /home/runner/Steam/ssfn..."
+echo "$ssfnFileContents" | base64 -d > "/home/runner/Steam/$ssfnFileName"
+chmod 777 "/home/runner/Steam/$ssfnFileName"
+
+echo "Finished Copying SteamGuard Files!"
+echo ""
+
+echo ""
+echo "#################################"
+echo "#        Uploading build        #"
+echo "#################################"
+echo ""
+
+$STEAM_CMD +login "$username" "$password" +run_app_build $(pwd)/manifest.vdf +quit || (
+    echo ""
+    echo "#################################"
+    echo "#             Errors            #"
+    echo "#################################"
+    echo ""
+    echo "Listing current folder and rootpath"
+    echo ""
+    ls -alh
+    echo ""
+    ls -alh $rootPath
+    echo ""
+    echo "Listing logs folder:"
+    echo ""
+    ls -Ralph "/home/runner/Steam/logs/"
+    echo ""
+    echo "Displaying error log"
+    echo ""
+    cat "/home/runner/Steam/logs/stderr.txt"
+    echo ""
+    echo "Displaying bootstrapper log"
+    echo ""
+    cat "/home/runner/Steam/logs/bootstrap_log.txt"
+    echo ""
+    echo "#################################"
+    echo "#             Output            #"
+    echo "#################################"
+    echo ""
+    ls -Ralph BuildOutput
+    exit 1
+  )
```

Comment:
Just curious. What made you decide to move all scripts into a single file and move away from docker?

Reply:
There were some differences in the paths used by the docker image vs the `CyberAndrii/setup-steamcmd` action. Since other people had successfully proven that `CyberAndrii/setup-steamcmd` works well, I thought it would be easier to just re-use that instead of trying to figure out which paths I needed to use with the docker image (I did initially spend some time trying to figure that out, but I got stuck and didn't want to spend more time going down that route). 
As for making it all a single script, that was because I was trying to reduce complexity as much as possible, and having to juggle multiple bash scripts added extra complexity I didn't want to deal with. I still think it's a good idea to move this logic to typescript eventually.

Reply:
I see. Yea makes total sense.

TypeScript seems the way to go for the future indeed.